engine/runtime/src/main/java/org/enso/compiler/pass/analyse/ExportSymbolAnalysis.java

```diff
@@ -0,0 +1,138 @@
+package org.enso.compiler.pass.analyse;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import org.enso.compiler.context.InlineContext;
+import org.enso.compiler.context.ModuleContext;
+import org.enso.compiler.core.IR;
+import org.enso.compiler.core.ir.Expression;
+import org.enso.compiler.core.ir.Module;
+import org.enso.compiler.core.ir.expression.errors.ImportExport;
+import org.enso.compiler.core.ir.module.scope.Export;
+import org.enso.compiler.data.BindingsMap;
+import org.enso.compiler.pass.IRPass;
+import org.enso.interpreter.util.ScalaConversions;
+import scala.collection.immutable.Seq;
+import scala.jdk.javaapi.CollectionConverters;
+
+/**
+ * This pass ensures that all the symbols that are exported exist. If not, an IR error is generated.
+ */
+public class ExportSymbolAnalysis implements IRPass {
+  public static final ExportSymbolAnalysis INSTANCE = new ExportSymbolAnalysis();
+  private UUID uuid;
+
+  private ExportSymbolAnalysis() {}
+
+  @Override
+  public UUID key() {
+    return null;
+  }
+
+  @Override
+  public void org$enso$compiler$pass$IRPass$_setter_$key_$eq(UUID v) {
+    this.uuid = v;
+  }
+
+  @Override
+  public Seq<IRPass> precursorPasses() {
+    List<IRPass> passes = List.of(
+        BindingAnalysis$.MODULE$,
+        ImportSymbolAnalysis$.MODULE$
+    );
+    return CollectionConverters.asScala(passes).toList();
```

Comment:
Yes, we shall compute this lazily or again and again.

Reply:
Oh, right. In Scala, this is `lazy val`. I introduced ` private static final` field in 430691420847f8bf62808384c42f09f53d38e8a0 to simulate that here, so that it's lazy as well.